packages/vincent-contracts/src/LibVincentDiamondStorage.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./diamond-base/libraries/LibDiamond.sol";
+import "./IPKPNftFacet.sol";
+
+library VincentAppStorage {
+    using EnumerableSet for EnumerableSet.UintSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 internal constant APP_STORAGE_SLOT = keccak256("lit.vincent.app.storage");
+
+    struct VersionedApp {
+        EnumerableSet.Bytes32Set toolIpfsCidHashes;
+        EnumerableSet.UintSet delegatedAgentPkps;
+        uint256 version;
+        bool enabled;
+    }
+
+    struct App {
+        EnumerableSet.AddressSet delegatees;
+        EnumerableSet.Bytes32Set authorizedDomains;
+        EnumerableSet.Bytes32Set authorizedRedirectUris;
+        VersionedApp[] versionedApps;
+        address manager;
+        string name;
+        string description;
+    }
+
+    struct AppStorage {
+        mapping(uint256 => App) appIdToApp;
+        mapping(address => EnumerableSet.UintSet) managerAddressToAppIds;
+        mapping(address => uint256) delegateeAddressToAppId;
+        mapping(bytes32 => string) authorizedDomainHashToDomain;
+        mapping(bytes32 => string) authorizedRedirectUriHashToRedirectUri;
+        EnumerableSet.UintSet registeredApps;
```

Comment:
Is enumeration necessary for this? I don't see any view function that requires going through the entire array / set.

Reply:
Removed in one of the latest commits. Replace with just the app ID counter uint256